ARKitExample/ViewController.swift

```diff
@@ -0,0 +1,218 @@
+                                                  nil,
+                                                  &pixelBuffer)
+
+        if (status != kCVReturnSuccess) {
+            return nil;
+        }
+
+        return pixelBuffer
+    }
+}
+
+// MARK: ARSCNViewDelegate
+extension ViewController: ARSCNViewDelegate {
+    func session(_ session: ARSession, didFailWithError error: Error) {
+        // Present an error message to the user
+        print("didFailWithError \(error)")
+    }
+
+    func sessionWasInterrupted(_ session: ARSession) {
+        // Inform the user that the session has been interrupted, for example, by presenting an overlay
+        print("ARSession was interrupted, disabling the VideoTrack.")
+        self.videoTrack?.isEnabled = false
+    }
+
+    func sessionInterruptionEnded(_ session: ARSession) {
+        // Reset tracking and/or remove existing anchors if consistent tracking is required
+        print("ARSession interruption ended, enabling the VideoTrack.")
+        self.videoTrack?.isEnabled = true
+    }
+}
+
+// MARK: TVIRoomDelegate
+extension ViewController: TVIRoomDelegate {
+    func didConnect(to room: TVIRoom) {
+        print("Connected to Room /(room.name).")
+    }
+
+    func room(_ room: TVIRoom, didFailToConnectWithError error: Error) {
+        print("Failed to connect to a Room: \(error).")
+
+        let alertController = UIAlertController.init(title: "Connection Failed",
+                                                     message: "Couldn't connect to Room \(room.name). code:\(error._code) \(error.localizedDescription)",
+                                                     preferredStyle: UIAlertControllerStyle.alert)
+
+        let cancelAction = UIAlertAction.init(title: "Okay", style: UIAlertActionStyle.default, handler: nil)
+        alertController.addAction(cancelAction)
+
+        self.present(alertController, animated: true, completion: nil)
+    }
+
+    func room(_ room: TVIRoom, participantDidConnect participant: TVIParticipant) {
+        print("Participant \(participant.identity) connected to \(room.name).")
+    }
+
+    func room(_ room: TVIRoom, participantDidDisconnect participant: TVIParticipant) {
+        print("Participant \(participant.identity) disconnected from \(room.name).")
+    }
+}
+
+// MARK: TVIVideoCapturer
+extension ViewController: TVIVideoCapturer {
+    var isScreencast: Bool {
+        // We want fluid AR content, maintaining the original frame rate.
+        return false
+    }
+
+    var supportedFormats: [TVIVideoFormat] {
+        // We only support the single capture format that ARSession provides, and we rasterize the AR scene at 1x.
+        // Don't set any specific capture dimensions.
+        let format = TVIVideoFormat.init()
+        format.frameRate = UInt(sceneView.preferredFramesPerSecond)
+        format.pixelFormat = TVIPixelFormat.format32BGRA
+        return [format]
```

Comment:
I was getting an invalid constraints error when specifying the screen size as the dimensions for the TVIVideoFormat. I will file a bug and look into this, but I don't think we should let it gate merging the PR.